chcemvediet/apps/accounts/models.py

```diff
@@ -32,11 +34,16 @@ class Profile(FormatMixin, models.Model):
                 If true, published inforequests will be shown anonymized, otherwise in their
                 original version.
                 """))
-    custom_anonymized_strings = JSONField(null=True, default=None,
+
+    # May be NULL; Should NOT be empty
+    custom_anonymized_strings = JSONField(null=True, blank=True, default=None,
             help_text=squeeze(u"""
-                User defined strings for anonymization. NULL for default anonymization.
+                User defined strings for anonymization. JSON must be an array of strings. NULL for
+                default anonymization.
                 """))
 
+    __original_custom_anonymized_strings = None
```

Comment:
@martinmacko47 Zistovanie ci sa `custom_anonymized_strings` zmenilo, treba overovat aj po zmene dat cez admina. Preto som logiku presunul do samotneho modelu v metode `save`.
Postupoval som podla navodu a nazov premennej zacinal s `__`, treba ju premenovat?

Reply:
@viliambalaz To vyzera ako feature, co mozno budeme potrebovat pouzit casom aj niekde inde a zislo by sa ju spravit vseobecne.

V `poleno/utils/models.py` sprav `OriginalValuesMixin`, co implementuje taketo odkladanie povodnych hodnot automaticky. Tj:
- V `__init__` ulozi vsetky fieldy do `self._originals` (alebo niekde tak)
- Na konci `save` aktualizuje hodnoty v `self._originals`
- A implementuje metodu `get_original_value(self, field)`, ktora vrati hodnotu zo `self._originals`

Potom v `Profile` uz len pouzijes `OriginalValuesMixin` a v `_diff_custom_anonymized_strings` ziskas povodnu hodnotu fieldu pomocou `self.get_original_value('custom_anonymized_strings')`

Btw, prefix `__foo` s dvomi `_` nepouzivaj. Python s dvomi `_` robi magiu, ktora sposobi, ze k danemu fieldu sa da pristupovat iba z danej classy. Dokonca sa k nim neda pristupovat ani z potomkov tej classy. Hoci teoreticky je to to, co chceme, ale prakticky je lepsie mat moznost ten field niekedy precitat (pri testovani, debuggovani a podobne). Preto sa privatne fieldy znacia zvycajne iba s jednym `_`.

Reply:
@viliambalaz Toto tiez nie je zapracovane